### DISCUSSION TITLE
date-picker enabling only specific dates selectable

### DISCUSSION THREAD

**Discussion by gaddoz** [vmware/clarity#5949] [General]
is it possible to restrict dates selectable from date-picker to a custom array of predefined dates?

**Comment by gaddoz** [comment]
@Jinnie

**Reply by bbogdanov** [reply]
@gaddoz If you are using `[ngModel]` you can use `[(ngModelChange])` for validate the change and update the value.

For `[formControl]` you can just `subscribe` to the `valueChanges`.

In both ways the end will be to just check if the new value exist in an array and if does not, then just keep the previous value on the control.

**Reply by gaddoz** [reply]
Thanks for your reply. It look great but is it possible to display and mark the available dates?

**Reply by bbogdanov** [reply]
We do not provide such a way at this stage.

It is possible to play around with DOM and find all `clr-day` components that are not holding the date you are looking for and append to the button inside each componentt a `is-excluded` class.

**Reply by gaddoz** [reply]
@bbogdanov thank you very much for your reply. I would like to follow your suggestion playing with DOM but I'm missing how to detect the date picker toggle event, do you have any idea?

**Reply by bbogdanov** [reply]
Unfortunately, we didn't expose the observable that you could use to `subscribe`. On the other hand you can select with `@ViewChild` the calendar button and add an `addEventListener`.

Something similar to this should work
```
`@ViewChild(`button.clr-smart-open-close`, { read: ElementRef }) datePickerCloseButton: HTMLButtonElement`

ngAfterViewInit() {
  if (this.datePickerCloseButton) {
    this.datePickerCloseButton.addEventLisnter('click', () => {
        // your logic
    })
  }
}
```
